### PR TITLE
Clippy allow single_element_loop, expect_fun_call in examples, bump h2

### LINKED ISSUE #1694: Clippy: specific allows for examples
Currently the examples have a lot of manual `#[allow(clippy::single_element_loop)]`
We could update when rust 1.75 is stable to use a `[lint]` section in `Cargo.toml`.
As this will cover the entire workspace member, we have to disallow the allowed lints in the sdk/src/lib.rs again.
(or wait until this works per folder)

An optional second task is update 
`std::env::var(var).unwrap_or_else(|_| panic!(".env variable '{var}' is undefined, see .env.example"));`
to
`std::env::var(var).expect(&format!(".env variable '{var}' is undefined, see .env.example"));`

which is currently not allowed due to `unwrap` 

issue stemming form #1676

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Clippy allow single_element_loop, expect_fun_call in examples, bump h2.

## Links to any relevant issues

Closes https://github.com/iotaledger/iota-sdk/issues/1694.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running `cargo clippy --examples`

### CODE REVIEW

File: sdk/src/lib.rs [hunk 1]
```diff
@@ -11,6 +11,8 @@
     clippy::significant_drop_in_scrutinee,
     clippy::significant_drop_tightening
 )]
+// Allowed in Cargo.toml for examples
+#![deny(clippy::single_element_loop, clippy::single_element_loop)]
```
Comment (thibault-martinez) [inline sdk/src/lib.rs]: :trollface:

Comment (Thoralf-M) [reply sdk/src/lib.rs]: Thanks

**Review by thibault-martinez — APPROVED** [review]

File: sdk/examples/client/01_generate_addresses.rs [hunk 2]
```diff
@@ -20,7 +20,7 @@ async fn main() -> Result<()> {
     dotenvy::dotenv().ok();
 
     for var in ["NODE_URL", "MNEMONIC"] {
-        std::env::var(var).unwrap_or_else(|_| panic!(".env variable '{var}' is undefined, see .env.example"));
+        std::env::var(var).expect(&format!(".env variable '{var}' is undefined, see .env.example"));
```
Comment (Alex6323) [inline sdk/examples/client/01_generate_addresses.rs:23]: not really sure I like it ... why not just?
```
`.unwrap_or_else(|| panic!(".env variable '{var}' is undefined, see .env.example"))
```
it's a bit longer and not exactly the same as expect, yes but wouldn't we wouldn't have to do the weird enable/disable lint :shrug: 

there really should be an `expect_with` or smth in std that takes a closure.

Comment (Thoralf-M) [reply sdk/examples/client/01_generate_addresses.rs:23]: > why not just? .... it's a bit longer

This was the only reason to have less distracting code in the examples, don't think the lints are that weird

But you can fight with wild @kwek20 about what's better :trollface:

Comment (thibault-martinez) [reply sdk/examples/client/01_generate_addresses.rs:23]: I don't really have a strong opinion but I'm willing to do slightly unconventional things if it makes the examples a bit more straight to the point.

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**